tests/unit/C/services/notification/test_data_availability_rule.cpp

```diff
@@ -0,0 +1,771 @@
+    // For now, we'll test with a simpler configuration
+    EXPECT_TRUE(triggers.find("AUDIT001") != string::npos);
+    EXPECT_TRUE(triggers.find("AUDIT002") != string::npos);
+    EXPECT_TRUE(triggers.find("alert") != string::npos);
+    
+    // Cleanup
+    rule.shutdown();
+}
+
+/**
+ * Test evaluation with invalid JSON
+ */
+TEST_F(DataAvailabilityRuleTest, EvalWithInvalidJSON)
+{
+    // Arrange
+    DataAvailabilityRule rule("TestDataAvailability");
+    ConfigCategory config = createBasicConfig("AUDIT001");
+    rule.init(config);
+    
+    // Act
+    bool result = rule.eval("invalid json");
+    
+    // Assert
+    EXPECT_FALSE(result);
+    
+    // Cleanup
+    rule.shutdown();
+}
+
+/**
+ * Test evaluation with empty JSON
+ */
+TEST_F(DataAvailabilityRuleTest, EvalWithEmptyJSON)
+{
+    // Arrange
+    DataAvailabilityRule rule("TestDataAvailability");
+    ConfigCategory config = createBasicConfig("AUDIT001");
+    rule.init(config);
+    
+    // Act
+    bool result = rule.eval("{}");
+    
+    // Assert
+    EXPECT_FALSE(result);
+    
+    // Cleanup
+    rule.shutdown();
+}
+
+/**
+ * Test evaluation with matching audit code
+ */
+TEST_F(DataAvailabilityRuleTest, EvalWithMatchingAuditCode)
+{
+    // Arrange
+    DataAvailabilityRule rule("TestDataAvailability");
+    ConfigCategory config = createBasicConfig("AUDIT001");
+    rule.init(config);
+    
+    string testJSON = createTestJSON("AUDIT001");
+    
+    // Act
+    bool result = rule.eval(testJSON);
+    
+    // Assert
+    EXPECT_TRUE(result);
+    
+    // Cleanup
+    rule.shutdown();
+}
+
+/**
+ * Test evaluation with non-matching audit code
+ */
+TEST_F(DataAvailabilityRuleTest, EvalWithNonMatchingAuditCode)
+{
+    // Arrange
+    DataAvailabilityRule rule("TestDataAvailability");
+    ConfigCategory config = createBasicConfig("AUDIT001");
+    rule.init(config);
+    
+    string testJSON = createTestJSON("DIFFERENT_AUDIT");
+    
+    // Act
+    bool result = rule.eval(testJSON);
+    
+    // Assert
+    EXPECT_FALSE(result);
+    
+    // Cleanup
+    rule.shutdown();
+}
+
+/**
+ * Test evaluation with timestamp
+ */
+TEST_F(DataAvailabilityRuleTest, EvalWithTimestamp)
+{
+    // Arrange
+    DataAvailabilityRule rule("TestDataAvailability");
+    ConfigCategory config = createBasicConfig("AUDIT001");
+    rule.init(config);
+    
+    string testJSON = createTestJSON("AUDIT001", 1234567890.123);
+    
+    // Act
+    bool result = rule.eval(testJSON);
+    
+    // Assert
+    EXPECT_TRUE(result);
+    
+    // Cleanup
+    rule.shutdown();
+}
+
+/**
+ * Test evaluation with multiple audit codes
+ */
+TEST_F(DataAvailabilityRuleTest, EvalWithMultipleAuditCodes)
+{
+    // Arrange
+    DataAvailabilityRule rule("TestDataAvailability");
+    ConfigCategory config = createBasicConfig("AUDIT001,AUDIT002");
+    rule.init(config);
+    
+    string testJSON = "{\"AUDIT001\": \"value1\", \"AUDIT002\": \"value2\"}";
```

Comment:
Can raw string literal with R"()" be used here?

Reply:
It could on those platforms that support it. We don't use it generally because we had issues on some platforms. Hence the use of QUOTE instead.

Reply:
No QUOTE has been used.